### DISCUSSION TITLE
Adding Labels to Self-Hosted Runners in GHA Runner Scale Set Chart

### DISCUSSION THREAD

**Discussion by omerap12** [community/community#135911] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello everyone,
When installing the `gha-runner-scale-set` chart (`oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set`), you can specify a `runnerScaleSetName`, such as `runnerScaleSetName: my-scale-set`, which creates a self-hosted runner with the name `my-scale-set`. 

Is there a way to add labels to this self-hosted runner? I'd like to include additional labels so that I can leverage them as described in the documentation: [Using labels to control where jobs are run](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/running-jobs-on-larger-runners#using-labels-to-control-where-jobs-are-run).

Any guidance on how to achieve this would be appreciated.

Thanks!

**Comment by nebuk89** [comment]
Not currently! We only support the runner names on the scale sets. I know this has caused a bunch of friction for folks, and we are trying to work out a path forward on our side. There are a ton of conversations around this in the Action-runner-controller repo. <3 
To be clear though, this isn't something we will be changing this calendar year :(

**Reply by omerap12** [reply, marked as answer]
Hey @nebuk89 , thanks for the reply.

**Comment by micha-6** [comment]
To add labels to self-hosted runners in the GitHub Actions Runner Scale Set Chart, modify the `values.yaml` file by adding your labels under the `runnerScaleSet.runners.labels` key:

```yaml
runnerScaleSet:
  runners:
    labels:
      - label1
      - label2
```Alternatively, use the command line to set labels during deployment:
`bash
helm upgrade --install my-runner-scale-set ghcr.io/actions/runner-controller/actions-runner-scale-set \set runnerScaleSet.runners .labels[0]=label1 \ set runner Scale Set . runners .labels[1]=label2 This ensures your runners are[ kinemaster mod apk](https://apkblings.com/kinemaster-mod-apk/) labeled appropriately for use in GitHub Actions workflows.

**Reply by omerap12** [reply]
That's odd because the comment above mentioned it's not possible. Regardless, could you please show me where it's located in the chart? I've checked [actions-runner-scale-set](https://github.com/actions/actions-runner-controller/tree/master/charts/gha-runner-scale-set/templates), but I can't seem to find it anywhere.

**Reply by omerap12** [reply]
It doesn't work

**Comment by HAMZA-420** [comment]
To add labels to a self-hosted runner when using the `gha-runner-scale-set` Helm chart, you can customize the Helm chart values to include labels for the runner. Here’s how you can do it:

### 1. **Modify the Helm Chart Values**
   When installing the `gha-runner-scale-set` chart, you can pass a `runnerLabels` value that includes the desired labels. This will attach the specified labels to the self-hosted runners in your scale set.

   You can achieve this by including the `runnerLabels` in your `values.yaml` or directly in the Helm install command.

   #### **Option 1: Using `values.yaml`**
   Add the `runnerLabels` under your `runner` configuration in your `values.yaml` file:

   ```yaml
   runnerScaleSetName: my-scale-set
   runner:
     runnerLabels:
       - label1
       - label2
       - label3
   ```

   #### **Option 2: Directly via Helm Command**
   Alternatively, you can specify the labels directly when you install the Helm chart:

   ```bash
   helm install my-scale-set oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set \
     --set runnerScaleSetName=my-scale-set \
     --set runner.runnerLabels[0]=label1 \
     --set runner.runnerLabels[1]=label2 \
     --set runner.runnerLabels[2]=label3
   ```

### 2. **Using Labels in Your Workflows**
   After adding these labels, you can control where jobs are run by specifying the labels in your GitHub Actions workflows:

   ```yaml
   jobs:
     build:
       runs-on: self-hosted,label1,label2
       steps:
         - name: Checkout code
           uses: actions/checkout@v2
         # Add other steps here
   ```

### 3. **Apply and Verify**
   - **Install or Upgrade**: Run the Helm install or upgrade command with your specified values.
   - **Check Labels**: Verify that the runners in the scale set have the correct labels by checking the runner configuration in GitHub under your repository or organization settings.

By following these steps, you can successfully add and use custom labels with your self-hosted runners created through the `gha-runner-scale-set` Helm chart.

**Reply by omerap12** [reply]
It doesn't work

**Comment by doug62** [comment]
Has this been fixed?  Is it truly not possible to use labels?
